posts/modular-monolith-microservices.md

```diff
@@ -0,0 +1,76 @@
+---
+authors:
+- Matthew Freshwaters
+tags:
+- entando
+- microservice architecture
+title: "Why Companies are Moving their Applications to Modular Architecture"
+image:
+---
+
+As applications become increasingly complex, engineering teams have sought to ease the process of creating scalable and malleable products. Many organizations have turned to microservices to help keep their products scalable, cost-efficient, and easier to develop over time. Today we will discuss why microservices & modular architecture have gained much attention and usage in the past few years.
+
+## Monolithic Architecture
+
+To understand microservice architecture better, let's take a look into what traditional application architecture looks like - the monolith. "A monolithic architecture is a singular, large computing network with one code base that couples all of the business concerns together. To make a change to this sort of application requires updating the entire stack by accessing the code base and building and deploying an updated version of the service-side interface." [Atlassian](https://www.atlassian.com/microservices/microservices-architecture/microservices-vs-monolith#:~:text=A%20monolithic%20architecture%20is%20a%20singular%2C%20large%20computing%20network%20with,of%20the%20service%2Dside%20interface.)
+
+Organizations and teams worldwide have approached application development with an agile mindset, hardening the process of developing a monolith. Making a single change to your application's server or client-side can become an extensive process, with having to roll out the entire application again. This resource-intensive activity can make CI/CD difficult, so application architects and developers have leaned into a different type of architecture - microservice architecture. We'll first take a look at the Modular Monolith and how it differs from a traditional monolith before moving into microservices.
+
+![Monolith Vs Microservice Architecture Diagram](../images/2022/12/mono-micro-diagram.png) ***Microservice vs Monolith Architecture***
+
+## Modular Monolith
+
+A Modular Monolith consists of dividing logic into modules, as each module is independent and isolated. Every module is in charge of its own business logic, database or schema, etc. With this model of architecture, you can build and modify existing layers of each module while leaving the others unaffected.
+
+Modules should communicate through APIs, allowing access to the logic of each module through public methods, not internal functions & the logic of each one. This architecture model is less complex than microservices and more scalable and easier to maintain than traditional monolithic apps. 
```

Comment:
Not necessarily - it has some advantages (if you have a very small application it might not make sense to design a full microservice architecture. While sharing some of the benefits of a microservice.  For example, a modular monolith will have to work under one tech stack whereas with microservices you could have a Spring microservice, a node / Express microservice, python based, etc.